test/email.routes.test.js

```diff
@@ -0,0 +1,161 @@
+  },
+  {
+    name: 'Aseem Sood',
+    email: 'asood123+test@gmail.com',
+    role: 'MEMBER'
+  },
+  {
+    name: 'Pia Mancini',
+    email: 'pia+test@opencollective.com',
+    role: 'BACKER',
+    lists: ['mailinglist.backers']
+  },
+  {
+    name: 'github',
+    email: 'github+test@opencollective.com',
+    role: 'BACKER',
+    lists: ['mailinglist.backers']
+  }
+]
+
+const groupData = {
+  slug: 'testcollective',
+  name: 'Test Collective',
+  settings: {}
+}
+
+let group;
+
+describe("controllers.services.email", () => {
+
+  let sandbox;
+
+  before((done) => utils.cleanAllDb().tap(a => done()));
+
+  after(() => nock.cleanAll());
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+  })
+
+  afterEach(() => {
+    nock.cleanAll()
+    sandbox.restore();
+  });
+
+  before((done) => {
+
+    Group.create(groupData)
+      .tap(g => group = g )
+      .then(() => User.createMany(usersData))
+      .tap(users => {
+        return Promise.map(users, (user, index) => {
+          return group.addUserWithRole(user, usersData[index].role);
+        });
+      })
+      .tap(users => {
+        return Promise.map(users, (user, index) => {
+          const lists = usersData[index].lists || [];
+          return Promise.map(lists, (list) => models.Notification.create({
+              channel: 'email',
+              UserId: user.id,
+              GroupId: group.id,
+              type: list
+            })
+          );
+        })
+      })
+      .then(() => done())
+      .catch(e => console.error);
+  });
+
+  it("forwards emails sent to info@:slug.opencollective.com", (done) => {
+
+    const spy = sandbox.spy(emailLib, 'sendMessage', (recipient, subject, body, opts) => {
+      return Promise.resolve();
+    });
+
+    request(app)
+      .post('/webhooks/mailgun')
+      .send(_.defaults({recipient: 'info@testcollective.opencollective.com'}, webhookBody))
+      .then(() => {
+        expect(spy.args[0][0]).to.equal('info@testcollective.opencollective.com');
+        expect(spy.args[0][1]).to.equal(webhookBody.subject);
+        expect(spy.args[0][3].bcc).to.equal(usersData[0].email);
+        done();
+      });
+  });
+  
+  it("forwards the email for approval to the core members", (done) => {
+
+    const spy = sandbox.spy(emailLib, 'send', (template, recipient, data) => {
+      return Promise.resolve();
+    });
+
+    request(app)
+      .post('/webhooks/mailgun')
+      .send(webhookBody)
+      .then(() => {
+        expect(spy.args[0][1]).to.equal('members@testcollective.opencollective.com');
+        const emailSentTo = [spy.args[0][3].bcc,spy.args[1][3].bcc];
+        expect(emailSentTo.indexOf(usersData[0].email) !== -1).to.be.true;
+        expect(emailSentTo.indexOf(usersData[1].email) !== -1).to.be.true;
+        done();
+      });
+  });
+
+  it("approves the email", (done) => {
+
+    const spy = sandbox.spy(emailLib, 'send', (template, recipient, data) => {
```

Comment:
I'm not sure that the third variable (the function) is needed here. Spies can't replace the original method, afaik. I used `const spy = sandbox.spy(emailLib, 'send')`.